src/world_interface/real_world_interface.cpp

```diff
@@ -96,33 +100,35 @@ void initMotors() {
 	}
 }
 
-void openCamera(CameraID camID, const char* cameraPath) {
+std::shared_ptr<cam::Camera> openCamera(CameraID camID, const std::string& cameraPath) {
```

Comment:
As I said, the Jetson can only open 2 cameras, so we're opening cameras as necessary (instead of at the beginning) and using reference counting to figure out when to close it. This involves weak and shared pointers, which should be pretty intuitive.